### ISSUE TITLE
 only supports 'NHWC' format under the GPU mode 

### ISSUE THREAD

**Issue by v-yunbin** [tensorflow/tensorflow#20405]
I use tensorflow 1.8 version.
we konw the gpu should support both 'NHWC' and 'NCHW' but i get bellow errors under GPU mode:

Conv2DCustomBackpropInputOp only supports NHWC.
	 [[Node: model_1/inference/conv2d_transpose/conv2d_transpose = Conv2DBackpropInput[T=DT_FLOAT, data_format="NCHW", dilations=[1, 1, 1, 1], padding="SAME", strides=[1, 1, 1, 16], **use_cudnn_on_gpu=true**, _device="/job:localhost/replica:0/task:0/device:CPU:0"](model_1/inference/conv2d_transpose/stack, model/inference/conv2d_transpose/kernel/read, model_1/inference/ExpandDims_1)]]

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by asimshankar** [comment]
CC @zheng-xq for comment, but I don't believe anyone is actively working on this at the moment, so marking as "Contributions Welcome".

**Comment by ppwwyyxx** [comment]
No this probably is a user error, because the op was placed on CPU:  `_device="/job:localhost/replica:0/task:0/device:CPU:0"`.
If the GPU is configured properly and the op is placed correctly on GPUs by users, the op `Conv2DCustomBackpropInputOp` will not appear in the graph at all.

**Comment by v-yunbin** [comment]
@ppwwyyxx I think i have a right config about gpu:
        num_gpus = 1, #Determines the number of gpus in use
	#Memory allocation on the memory
	config = tf.ConfigProto()
	config.gpu_options.allow_growth = True

	#Train
	with tf.Session(config=config) as sess:
		try:
			summary_writer = tf.summary.FileWriter(log_dir, sess.graph)
			sess.run(tf.global_variables_initializer())

**Comment by jvishnuvardhan** [comment]
This is a stale issue. Looks like in the recent TF versions, other data formats also supported. For example, check [this page](https://www.tensorflow.org/api_docs/python/tf/nn/depth_to_space).

> # tf.nn.depth_to_space
> The data_format attr specifies the layout of the input and output tensors with the following options: "NHWC": [ batch, height, width, channels ] "NCHW": [ batch, channels, height, width ] "NCHW_VECT_C": qint8 [ batch, channels / 4, height, width, 4 ]
> 

Closing this issue as this was resolved in recent TF versions. Feel free to reopen if I am mistaken. Thanks!
